sdk/src/wallet/account/operations/output_consolidation.rs

```diff
@@ -143,28 +153,32 @@ impl Account {
             });
         }
 
-        let max_inputs = match &*self.wallet.secret_manager.read().await {
-            #[cfg(feature = "ledger_nano")]
-            SecretManager::LedgerNano(ledger) => {
-                let ledger_nano_status = ledger.get_ledger_nano_status().await;
-                // With blind signing we are only limited by the protocol
-                if ledger_nano_status.blind_signing_enabled() {
-                    INPUT_COUNT_MAX
-                } else {
-                    ledger_nano_status
-                        .buffer_size()
-                        .map(|buffer_size| {
-                            // Calculate how many inputs we can have with this ledger, buffer size is different for
-                            // different ledger types
-                            let available_buffer_size_for_inputs =
-                                buffer_size - ESSENCE_SIZE_WITHOUT_IN_AND_OUTPUTS - MIN_OUTPUT_SIZE_IN_ESSENCE;
-                            (available_buffer_size_for_inputs / INPUT_SIZE) as u16
-                        })
-                        .unwrap_or(INPUT_COUNT_MAX)
-                }
+        #[cfg(feature = "ledger_nano")]
+        let max_inputs = if let Some(ledger) = ((&*self.wallet.secret_manager.read().await)
+            as &(dyn std::any::Any + Send + Sync))
+            .downcast_ref::<crate::client::secret::ledger_nano::LedgerSecretManager>()
```

Comment:
This downcasting check is done in several places, so it might be worth moving it into a method `fn is_ledger_secret_manager(&self) -> bool`, maybe attributed with an inline or smth. Might improve readability of the code a bit.

Reply:
I think I would also do cast methods, probably `as_*`